Remove dependency on pytz

Jenselme opened the pull request:
Starting with Python 3.9 a zoneinfo package is available in the standard library for this feature. There is therefore no need for this dependency and a compatibility layer is available at
https://pypi.org/project/backports.zoneinfo/ for older Python versions.

I made this change because agate is packaged in Fedora and Fedora is deprecating pytz: https://fedoraproject.org/wiki/Changes/DeprecatePytz

jpmckinney commented:
@Jenselme Can you update the PR to use https://pypi.org/project/tzdata/?

As mentioned in [Python's docs](https://docs.python.org/3/library/zoneinfo.html#data-sources), Windows does "not have an IANA database available", and sure enough agate's tests fail on Windows.

Jenselme commented:
@jpmckinney I rebased my branch on `master` and added `tzdata` as a dependency for Windows. I guess I cannot test until you approve the workflow. Let me know if I need to make anything more!

Jenselme commented:
It seems fine on my end: https://github.com/Jenselme/agate/actions/runs/6441823659 & https://github.com/Jenselme/agate/actions/runs/6441823657

jpmckinney commented:
Thank you! I’ll make a release when I’m at my desk.

jpmckinney commented:
Released! 🚀
